neqo-common/src/timer.rs

```diff
@@ -111,6 +111,7 @@ impl<T> Timer<T> {
     /// Asserts if the time given is in the past or too far in the future.
     /// # Panics
     /// When `time` is in the past relative to previous calls.
+    #[allow(clippy::unchecked_duration_subtraction)] // guarded by assertion
```

Comment:
we've already checked that `time` is greater than `short_span`, so there is no need to check the duration subtraction.